x/auth/ante/basic.go

```diff
@@ -27,6 +29,12 @@ func (vbd ValidateBasicDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulat
 		return next(ctx, tx, simulate)
 	}
 
+	for _, msg := range tx.GetMsgs() {
+		if err := txvalidate.ValidateMsg(msg); err != nil {
+			return ctx, err
```

Comment:
@atheeshp Sorry my bad, I thought ValidateBasic for msgs was run in this antehandler, but it isnt, it's run in baseapp:
https://github.com/cosmos/cosmos-sdk/blob/cd221680c0509a39b539d3dd7e3e45d700aa4b54/baseapp/baseapp.go#L626-L629

And i think we should run ValidateMsgs there, instead of in the antehandler.

Reply:
However, there might be a dependency issue: we don't want baseapp to depend on client/tx, and we don't want client/tx to depend on baseapp. Do you have an idea where to put ValidateMsgs?

Reply:
yeah, thank you for spending time on this